sequence-model-generator/modules/sequence-model-generator-core/src/main/java/io/ballerina/sequencemodelgenerator/core/ParticipantBodyAnalyzer.java

```diff
@@ -0,0 +1,249 @@
+ *
+ * @since 2201.9.0
+ */
+public class ParticipantBodyAnalyzer extends NodeVisitor {
+
+    private final List<SequenceNode> sequenceNodes;
+    private final SemanticModel semanticModel;
+    private final Stack<SequenceNode.Builder> nodeBuilderStack;
+    private SequenceNode.Builder nodeBuilder;
+    private Node variableNode;
+
+    public ParticipantBodyAnalyzer(SemanticModel semanticModel) {
+        this.semanticModel = semanticModel;
+        this.sequenceNodes = new ArrayList<>();
+        this.nodeBuilderStack = new Stack<>();
+        this.nodeBuilder = new SequenceNode.Builder(semanticModel);
+    }
+
+    @Override
+    public void visit(RemoteMethodCallActionNode remoteMethodCallActionNode) {
+        String targetId = ParticipantManager.getInstance().getParticipantId(remoteMethodCallActionNode.expression());
+
+        nodeBuilder = new Interaction.Builder(semanticModel)
+                .interactionType(Interaction.InteractionType.ENDPOINT_CALL)
+                .targetId(targetId)
+                .location(remoteMethodCallActionNode);
+
+        nodeBuilder
+                .property(Interaction.PARAMS_LABEL, getParamList(remoteMethodCallActionNode.arguments()))
+                .property(Interaction.NAME_LABEL,
+                        Expression.Factory.createStringType(remoteMethodCallActionNode.methodName()))
+                .property(Interaction.EXPRESSION_LABEL, remoteMethodCallActionNode.expression())
+                .property(Interaction.VALUE_LABEL, Expression.Factory.create(semanticModel,
+                        remoteMethodCallActionNode, variableNode));
+
+        appendNode();
+    }
+
+    @Override
+    public void visit(VariableDeclarationNode variableDeclarationNode) {
+        this.variableNode = variableDeclarationNode.typedBindingPattern().bindingPattern();
+        variableDeclarationNode.initializer().ifPresent(expressionNode -> expressionNode.accept(this));
+        this.variableNode = null;
+    }
+
+    @Override
+    public void visit(AssignmentStatementNode assignmentStatementNode) {
+        this.variableNode = assignmentStatementNode.varRef();
+        super.visit(assignmentStatementNode);
+        this.variableNode = null;
+    }
+
+    @Override
+    public void visit(FunctionCallExpressionNode functionCallExpressionNode) {
+        NameReferenceNode functionName = functionCallExpressionNode.functionName();
+
+        String targetId = ParticipantManager.getInstance().getParticipantId(functionName);
+        nodeBuilder = new Interaction.Builder(semanticModel)
+                .interactionType(Interaction.InteractionType.FUNCTION_CALL)
+                .targetId(targetId)
+                .location(functionCallExpressionNode);
+
+        nodeBuilder
+                .property(Interaction.PARAMS_LABEL, getParamList(functionCallExpressionNode.arguments()))
+                .property(Interaction.NAME_LABEL, Expression.Factory.createStringType(functionName));
```

Comment:
why aren't we attaching a `EXPRESSION_LABEL` to this builder?

Reply:
EXPRESSION_LABEL refers to the expression in the following grammar rule: https://ballerina.io/spec/lang/master/#section_7.9.1, and the function call does not have one.